src/main/java/org/shanerx/tradeshop/utils/configuration/YAMLConfigurationController.java

```diff
@@ -0,0 +1,248 @@
+/*
+ *
+ *                         Copyright (c) 2016-2019
+ *                SparklingComet @ http://shanerx.org
+ *               KillerOfPie @ http://killerofpie.github.io
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *                http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  NOTICE: All modifications made by others to the source code belong
+ *  to the respective contributor. No contributor should be held liable for
+ *  any damages of any kind, whether be material or moral, which were
+ *  caused by their contribution(s) to the project. See the full License for more information.
+ *
+ */
+
+package org.shanerx.tradeshop.utils.configuration;
+
+import com.google.common.base.Charsets;
+import com.google.common.collect.Sets;
+import com.google.common.io.Files;
+import org.apache.commons.lang.Validate;
+import org.bukkit.configuration.file.FileConfiguration;
+import org.bukkit.configuration.file.YamlConfiguration;
+import org.shanerx.tradeshop.TradeShop;
+import org.shanerx.tradeshop.utils.configuration.interfaces.YAMLConfigurationInterface;
+import org.shanerx.tradeshop.utils.configuration.interfaces.YAMLConfigurationSectionInterface;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Level;
+
+public class YAMLConfigurationController {
+
+    protected final TradeShop plugin;
+    protected final File file;
+    protected FileConfiguration config;
+
+    public YAMLConfigurationController(TradeShop plugin, String fileNameNoExtension) {
+        file = new File(plugin.getDataFolder(), fileNameNoExtension + ".yml");
+        config = YamlConfiguration.loadConfiguration(file);
+        this.plugin = plugin;
+    }
+
+    public void addKey(String path, Object value) {
+        if (config.get(path) == null) {
+            config.set(path, value);
+        }
+    }
+
+    public void load() {
```

Comment:
Have all these IO methods been tested?

Reply:
Anything loading was copied from existing code and just re-organized

Reply:
Alright fair enough